client/src/components/uploadStoryForm/index.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable */
+import React, { useState } from 'react';
+import { Modal, Button, Input } from 'antd';
+import './styles.css';
+import httpService from '../../services/httpService';
+// import axios from 'axios';
+
+const { TextArea } = Input;
+
+function StoryForm() {
+  const [fileInputState, setFileInputState] = useState('');
+  const [selectedFile, setSelectedFile] = useState();
+  const [previewSource, setPreviewSource] = useState('');
+  const [visible, setVisible] = useState(false);
+  const [story, setStory] = useState({
+    title: null,
+    content: null,
+    data: null,
+  });
+
+  const previewFile = (file) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onloadend = () => {
+      setPreviewSource(reader.result);
+    };
+  };
+
+  const handleFileInputChange = (e) => {
+    const file = e.target.files[0];
+    previewFile(file);
+    setSelectedFile(file);
+    setFileInputState(e.target.value);
+  };
+
+  const uploadImage = async (base64EncodedImage) => {
+    try {
+      await httpService.post('/api/v1/story', {
+        data: base64EncodedImage,
+          title: story.title,
+          content: story.content,
+      },
+       { headers: { 'Content-Type': 'application/json' }}
+      );
+      setFileInputState('');
+      setPreviewSource('');
+    } catch (err) {
+      throw new Error(`Error ${err}`);
+    }
+  };
+  const handleSubmitFile = (e) => {
+    e.preventDefault();
+    if (!selectedFile) return;
+    const reader = new FileReader();
+    reader.readAsDataURL(selectedFile);
+    setVisible(false);
+    reader.onloadend = () => {
+      uploadImage(reader.result);
+    };
+  };
+
+  return (
+    <>
+      <Button type="primary" onClick={() => setVisible(true)}>
+        أضف قصتك
+      </Button>
+      <Modal
+        title="اضف قصتك"
+        centered
+        visible={visible}
+        onOk={handleSubmitFile}
+        okText="موافق"
+        cancelText="الغاء"
+        onCancel={() => setVisible(false)}
+        width={500}
+      >
+        <Input
+          onChange={(e) => setStory((prev) => ({ ...prev, title: e.target.value }))}
+          placeholder="العنوان"
+        />
+        <>
+          <TextArea
+            rows={4}
+            placeholder="القصة"
+            onChange={(e) => setStory((prev) => ({ ...prev, content: e.target.value }))}
+          />
+          <form onSubmit={handleFileInputChange} className="form">
```

Comment:
the form will wrap all our inputs not only file input, although notice that content and title are required in database so those values shouldn't be empty.
As we validate the request body on the server side we need to validate input's here else.
So, If I'm a user trying to submit form with empty values I should see an error message clarify that to me.
https://ant.design/components/form/ check the first example here to add rule to your input

Reply:
form onSubmit here will never work,
Instead make a ref to your form and handle onSubmit modal to run this function